src/feature_extraction.py

```diff
@@ -477,6 +477,112 @@ def persist(self, filepath):
+                            inplace=True)
+        features = features.merge(group_object, on=['SK_ID_CURR'], how='left')
+
+        group_object = prev_applications_last_5.groupby(by=['SK_ID_CURR'])['CNT_PAYMENT'].mean().reset_index()
+        group_object.rename(index=str,
+                            columns={'CNT_PAYMENT': 'previous_application_term_of_last_5_credit_mean'},
+                            inplace=True)
+        features = features.merge(group_object, on=['SK_ID_CURR'], how='left')
+
+        group_object = prev_applications_sorted.groupby(by=['SK_ID_CURR'])['DAYS_DECISION'].last().reset_index()
+        group_object.rename(index=str,
+                            columns={'DAYS_DECISION': 'previous_application_days_decision_about_prev_credit'},
+                            inplace=True)
+        features = features.merge(group_object, on=['SK_ID_CURR'], how='left')
+
+        group_object = prev_applications_last_3.groupby(by=['SK_ID_CURR'])['DAYS_DECISION'].mean().reset_index()
+        group_object.rename(index=str,
+                            columns={'DAYS_DECISION': 'previous_application_days_decision_about_last_3_credit_mean'},
+                            inplace=True)
+        features = features.merge(group_object, on=['SK_ID_CURR'], how='left')
+
+        group_object = prev_applications_last_5.groupby(by=['SK_ID_CURR'])['DAYS_DECISION'].mean().reset_index()
+        group_object.rename(index=str,
+                            columns={'DAYS_DECISION': 'previous_application_days_decision_about_last_5_credit_mean'},
+                            inplace=True)
+        features = features.merge(group_object, on=['SK_ID_CURR'], how='left')
+
+        group_object = prev_applications_sorted.groupby(by=['SK_ID_CURR'])['DAYS_FIRST_DRAWING'].last().reset_index()
+        group_object.rename(index=str, columns={
+            'DAYS_FIRST_DRAWING': 'previous_application_days_first_disbursement_of_prev_application'}, inplace=True)
+        features = features.merge(group_object, on=['SK_ID_CURR'], how='left')
+
+        group_object = prev_applications_last_3.groupby(by=['SK_ID_CURR'])['DAYS_FIRST_DRAWING'].mean().reset_index()
+        group_object.rename(index=str, columns={
+            'DAYS_FIRST_DRAWING': 'previous_application_days_first_disbursement_last_3_credit_mean'}, inplace=True)
+        features = features.merge(group_object, on=['SK_ID_CURR'], how='left')
+
+        group_object = prev_applications_last_5.groupby(by=['SK_ID_CURR'])['DAYS_FIRST_DRAWING'].mean().reset_index()
+        group_object.rename(index=str, columns={
+            'DAYS_FIRST_DRAWING': 'previous_application_days_first_disbursement_last_5_credit_mean'}, inplace=True)
+        features = features.merge(group_object, on=['SK_ID_CURR'], how='left')
+
+        prev_applications_sorted['previous_application_prev_was_approved'] = (
+                prev_applications_sorted['NAME_CONTRACT_STATUS'] == 'Approved').astype('int')
```

Comment:
@pknut do we have features for fraction of approved applications or maybe if someone had applications approved in the past but lately he has been denied with higher frequency that could be a very strong feature.